bottom-navigator/src/main/java/com/pandora/bottomnavigator/FragmentTransactionHandler.kt

```diff
@@ -236,19 +245,63 @@ internal data class TagStructure private constructor(
         private const val OURTAG = "com.pandora.navigator"
         private const val SEPARATOR = "|"
         private const val DETACHABLE = "DETACHABLE"
+        private const val TRANSITIONS = "TRANSITIONS"
 
         fun fromTag(tag: String?): TagStructure {
             if (tag == null || !tag.startsWith(OURTAG)) {
-                return TagStructure(null, null, null)
+                return TagStructure(null, null, null, null)
                     .apply { isOurFragment = false }
             }
 
-            val (ourTag, className, detachable, uuid) = tag.split(SEPARATOR)
+            val (tagData, transitions) = TransitionsData.fromTag(tag)
+            val (ourTag, className, detachable, uuid) = tagData.split(SEPARATOR)
 
-            if (ourTag != OURTAG) return TagStructure(null, null, null)
+            if (ourTag != OURTAG) return TagStructure(null, null, null, null)
                 .apply { isOurFragment = false }
 
-            return TagStructure(className, DETACHABLE == detachable, uuid)
+            return TagStructure(className, DETACHABLE == detachable, uuid, transitions)
+        }
+    }
+
+    /**
+     * Data class for a fragment's transitions. Responsible for serializing and de-serializing itself
+     */
+    internal data class TransitionsData(
```

Comment:
TransitionsData can serialize itself with toString() and deserialize itself from a tag by extracting the transitions and returning it as a pair with the remaining tag. The motivation was to keep transition-specific fields out of TagStructure, but let me know if you think it'd be better to just serialize it all in TagStructure.toString()